rmw_connext_shared_cpp/src/types/custom_publisher_listener.cpp

```diff
@@ -23,7 +23,12 @@
 void CustomPublisherListener::on_data_available(DDS::DataReader * reader)
 {
   DDS::PublicationBuiltinTopicDataDataReader * builtin_reader =
-    static_cast<DDS::PublicationBuiltinTopicDataDataReader *>(reader);
+    DDS::PublicationBuiltinTopicDataDataReader::narrow(reader);
+
+  if (!builtin_reader) {
+    fprintf(stderr, "failed to narrow to DDS::PublicationBuiltinTopicDataDataReader\n");
```

Comment:
Recommend `RCUTILS_LOG_ERROR_NAMED` instead of `fprintf` as done here:

https://github.com/ros2/rmw_connext/blob/aa5f47f6af79d2d4846dffdbf0394a6eaaaf80d6/rmw_connext_shared_cpp/src/node_names.cpp#L135-L137

That will default to printing to stderr, but allow someone to change where log messages are output.

Reply:
Created new issue #344 to fix all printf through a separate issue

Reply:
Works for me. Thanks for the issue and PR!